src/identity-providers/add/AddMapper.tsx

```diff
@@ -54,11 +56,21 @@ export const AddMapper = () => {
   const { providerId, alias } = useParams<IdentityProviderAddMapperParams>();
   const { id } = useParams<IdentityProviderEditMapperParams>();
 
-  const isSAMLorOIDC = providerId === "saml" || providerId === "oidc";
+  const identityProviders = _.groupBy(
+    useServerInfo().identityProviders,
+    "groupName"
+  );
+
+  const isSocialIdP = identityProviders["Social"]
+    .map((item) => item.id)
+    .includes(providerId.toLowerCase());
 
   const [mapperTypes, setMapperTypes] =
     useState<Record<string, IdentityProviderMapperRepresentation>>();
-  const [mapperType, setMapperType] = useState("advancedAttributeToRole");
+  const [mapperType, setMapperType] = useState(
+    !isSocialIdP ? "hardcodedRole" : "attributeImporter"
```

Comment:
If you invert the order of this statement there is no need to negate the value.
```suggestion
    isSocialIdP ? "attributeImporter" : "hardcodedRole"
```